### DISCUSSION TITLE
Llama multi GPU

### DISCUSSION THREAD

**Discussion by PaulaScholz** [ggml-org/llama.cpp#3804] [Show and tell]
I have Llama2 running under LlamaSharp (latest drop, 10/26) and CUDA-12.  I took a screen capture of the Task Manager running while the model was answering questions and thought I'd provide you the feedback.  There are 4 A6000 GPUs on the system with 128GB of system ram.  It works, and also loads and runs the 70b models (albeit a bit more slowly).  Though it does use all the GPUs, it mostly puts the burden on GPU0.

I wanted to upload a larger video file, but the limit is 10mb.

https://github.com/ggerganov/llama.cpp/assets/45177251/12f269f4-820d-454b-8806-f234391b0a71

**Comment by quarterturn** [comment]
Something changed within the last week. I see this too on my 3x P40 setup, it is trying to utilize GPU0 almost by itself and I eventually get an OOM on the first prompt.

**Comment by dinorocks** [comment]
I have two Tesla P40s, and the primary GPU always gets most of the work. Not sure why. Could someone help? I remember it wasn't always this way, but couldn't pinpoint a moment in time when it changed. 
<img width="649" alt="image" src="https://github.com/ggerganov/llama.cpp/assets/96495700/38767eb1-f371-4956-8244-778dbbd69be2">
